feature/experimental/gribi/otg_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,736 @@
+				WithPrefix(virtualVIPs[ip]+"/32").
+				WithNetworkInstance(*deviations.DefaultNetworkInstance).
+				WithNextHopGroup(uint64(nhgID)).
+				WithElectionID(args.electionID.Low, args.electionID.High))
+	}
+
+	if err := awaitTimeout(args.ctx, args.client, t, time.Minute); err != nil {
+		t.Fatalf("Could not program entries via clientA, got err: %v", err)
+	}
+
+	if switchover {
+		for _, v := range args.client.Results(t) {
+			if v.ProgrammingResult == aftspb.AFTResult_FIB_PROGRAMMED {
+				fibProgrammedEntries = append(fibProgrammedEntries, v.Details.IPv4Prefix)
+			}
+		}
+	} else {
+		for ip := range virtualVIPs {
+			chk.HasResult(t, args.client.Results(t),
+				fluent.OperationResult().
+					WithIPv4Operation(virtualVIPs[ip]+"/32").
+					WithOperationType(constants.Add).
+					WithProgrammingResult(fluent.InstalledInFIB).
+					AsResult(),
+				chk.IgnoreOperationID(),
+			)
+		}
+
+		gr, err := args.client.Get().
+			WithNetworkInstance(*deviations.DefaultNetworkInstance).
+			WithAFT(fluent.IPv4).
+			Send()
+		if err != nil {
+			t.Fatalf("Error encountered during gRIBI Get operation: got: %v", err)
+		}
+
+		for ip := range virtualVIPs {
+			chk.GetResponseHasEntries(t, gr,
+				fluent.IPv4Entry().
+					WithNetworkInstance(*deviations.DefaultNetworkInstance).
+					WithNextHopGroup(uint64(nhgID)).
+					WithPrefix(virtualVIPs[ip]+"/32"),
+			)
+		}
+	}
+}
+
+// pushConfig pushes the configuration to the device using gNMI Replace.
+func pushConfig(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, d *oc.Root) {
+	t.Helper()
+
+	intfName := dutPort.Name()
+	i := d.GetOrCreateInterface(intfName)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(intfName).Config(), i)
+}
+
+// configureInterfaceDUT configures a single DUT layer 2 port.
+func configureInterfaceDUT(t *testing.T, dutPort *ondatra.Port, d *oc.Root, desc string) {
+	t.Helper()
+
+	i := d.GetOrCreateInterface(dutPort.Name())
+	i.Description = ygot.String(desc)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+	t.Logf("DUT port %s configured", dutPort)
+}
+
+// generateSubIntfPair configures ATE/DUT SubInterfaces on the target device and returns
+// a slice of the corresponding ATE IPAddresses.
+func generateSubIntfPair(t *testing.T, dut *ondatra.DUTDevice, dutPort *ondatra.Port, ate *ondatra.ATEDevice, atePort *ondatra.Port, top gosnappi.Config, d *oc.Root) []string {
+	t.Helper()
+
+	nextHops := []string{}
+	nextHopCount := 63 // nextHopCount specifies number of nextHop IPs needed.
+	top.Ports().Add().SetName(atePort.ID())
+	for i := 0; i <= nextHopCount; i++ {
+		vlanID := uint16(i) + 1
+		name := fmt.Sprintf(`dst%d`, i)
+		Index := uint32(i) + 1
+		ateIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 1))
+		dutIPv4 := fmt.Sprintf(`198.51.100.%d`, ((4 * i) + 2))
+		configureSubinterfaceDUT(t, d, dutPort, Index, vlanID, dutIPv4)
+		MAC, _ := incrementMAC(atePort1.MAC, i+1)
```

Comment:
Check the error/fail here if there is one.

Reply:
Done